angular-app/src/app/components/home/home.component.ts

```diff
@@ -93,39 +93,36 @@ export class HomeComponent implements OnInit {
   }
 
   /*  Get data for d3 chart. Sets local `nodes` and `links` values. 
-      Uses `netMetricsService.retrievePersistedDataForGraph` w `GET` 
-      request to api. 
-      Called at `setInterval` by `updateMetrics`. 
+      Uses `netMetricsService.retrievePersistedDataForGraph`. 
   */
   public initGraphData() {
     const that = this;
-    that.netMetricsService.retrievePersistedDataForGraph()
-      .then((res: NMResponse) => {
-        const persistedData = res.body;
-        // console.log('initGraphData: persistedData', persistedData);
-        const psuedoNodes = persistedData['nodes'];
-        const pseudoLinks = persistedData['links'];
-        that.nodes = [];
-        that.links = [];
-        // Instantiate real Node insances iso literal object:
-        for (const pseudoNode of psuedoNodes) {
-          const node = new Node(pseudoNode['id']);
-          // const node = new Node(pseudoNode['address']); // Here's where the bug is at: 'id' iso 'address'
-          node.x = Math.floor(Math.random() * 600) + 100;
-          node.y = Math.floor(Math.random() * 600) + 100;
-          node.linkCount = pseudoNode['numChannels'];
-          that.nodes.push(node);
-        }
-        // Get the real Node instance iso literal object:
-        for (const pseudoLink of pseudoLinks) {
-          // const link = new Link(pseudoLink['source'], pseudoLink['target']);
-          const link = new Link(that.getMatchingNode(pseudoLink['source'], that.nodes), that.getMatchingNode(pseudoLink['target'], that.nodes));
-          that.links.push(link);
-        }
-      })
-      .catch((err: any) => {
-        console.log('initGraphData error:', err);
-      });
+    const persistedData: NMRestructuredData = that.netMetricsService.retrievePersistedDataForGraph();
+    // console.log('initGraphData: persistedData', persistedData);
+    const psuedoNodes = persistedData['nodes'];
+    const pseudoLinks = persistedData['links'];
+    that.nodes = [];
+    that.links = [];
+    // Instantiate real Node instances iso literal object:
+    for (const pseudoNode of psuedoNodes) {
+      const node = new Node(pseudoNode['id']);
+      // const node = new Node(pseudoNode['address']); // Here's where the bug is at: 'id' iso 'address'
+      node.x = Math.floor(Math.random() * 600) + 100;
```

Comment:
So positions are random at the moment: do you think we can implement something like https://bl.ocks.org/mbostock/4062045

Reply:
Sure. I see a lot of potential for improving the d3 chart. I suggest later - not in this PR.

Reply:
See also #11 and #16